libs/arrow-spi/build.gradle

```diff
@@ -10,22 +10,10 @@
  */
 
 testingConventions.enabled = false
+
 dependencies {
   api project(':libs:opensearch-core')
-  api "org.apache.arrow:arrow-vector:${versions.arrow}"
-  api "org.apache.arrow:arrow-format:${versions.arrow}"
-  api "org.apache.arrow:arrow-memory-core:${versions.arrow}"
-  runtimeOnly "org.apache.arrow:arrow-memory-netty-buffer-patch:${versions.arrow}"
-  runtimeOnly "org.apache.arrow:arrow-memory-netty:${versions.arrow}"
-  runtimeOnly "io.netty:netty-buffer:${versions.netty}"
-  runtimeOnly "io.netty:netty-common:${versions.netty}"
-
-  runtimeOnly "com.google.flatbuffers:flatbuffers-java:${versions.flatbuffers}"
-  runtimeOnly "org.slf4j:slf4j-api:${versions.slf4j}"
-  runtimeOnly "com.fasterxml.jackson.core:jackson-databind:${versions.jackson}"
-  api "com.fasterxml.jackson.core:jackson-annotations:${versions.jackson}"
-
-  implementation "commons-codec:commons-codec:${versions.commonscodec}"
+  api project(path:':libs:opensearch-arrow-memory-shaded', configuration: 'shadow')
```

Comment:
So this is the only dependencies that SPI layer needs:

```
dependencies {
  api project(':libs:opensearch-core')
  api "org.apache.arrow:arrow-memory-core:${versions.arrow}"
  api "org.apache.arrow:arrow-vector:${versions.arrow}"
}

```

Reply:
Right. But at runtime, there are other transitive dependencies needed like netty-buffer, jackson-annotations etc which are part of `arrow-memory-netty-shaded` as shaded dependencies.

Reply:
> So if we have to use vector in server module, then these runtime dependencies should be present on its runtime classpath.

We will not use vector in server module, the server has to be sufficient with SPI layer at the moment (since everything else is in plugin)